### PR TITLE
Docs: Update ESP8266 intro to add Getting the firmware section

### PR DESCRIPTION
**Pull request by gbaman**
Add a "Getting the firmware" section to better describe how to get hold of the MicroPython firmware, especially if you have a 512kb module.

### CODE REVIEW

File: docs/esp8266/tutorial/intro.rst [hunk 1]
```diff
@@ -35,6 +35,25 @@ If your board has a USB connector on it then most likely it is powered through
...
+* Stable firmware builds for 1024kb modules and above.
+* Daily firmware builds for 1024kb modules and above.
+* Daily firmware builds for 512kb modules.
+
+The best bet is nearly always to go for the Stable firmware builds.
+An exception to this though is if you have an ESP8266 module with only 512kb
+of onboard storage. You can easily tell by trying to load a Stable firmware 
+build and if you get the error below, then you may have to use the Daily 
+firmware builds for 512kb modules.
+    WARNING: Unlikely to work as data goes beyond end of flash.
```
Comment (dpgeorge) [inline docs/esp8266/tutorial/intro.rst]: Is this warning emitted by esptool?

File: docs/esp8266/tutorial/intro.rst [hunk 2]
```diff
@@ -35,6 +35,25 @@ If your board has a USB connector on it then most likely it is powered through
...
+
```
Comment (dpgeorge) [inline docs/esp8266/tutorial/intro.rst]: In the paragraph below, the first sentence will need to be changed to something like "Once you have the firmware...".

File: docs/esp8266/tutorial/intro.rst (hunk 1 above)
Comment (gbaman) [inline docs/esp8266/tutorial/intro.rst]: Yip https://github.com/espressif/esptool/blob/0a6db9e7a0021cf43e5ef44134741e9cd577a269/esptool.py#L888

File: docs/esp8266/tutorial/intro.rst (hunk 2 above)
Comment (gbaman) [inline docs/esp8266/tutorial/intro.rst]: Updated.

**Comment by dpgeorge** [comment]
Thanks, merged in 71ff0b549d179acb210b4a2cc44ba726f61eede4.

**Pull request closed without merging**